src/VSPackage.cs

```diff
@@ -92,9 +93,14 @@ private async Task<bool> IsSolutionLoadedAsync()
             return value is bool isSolOpen && isSolOpen;
         }
 
-        private async void HandleOpenSolution(object sender = null, EventArgs e = null)
+        private void HandleOpenSolution(object sender, EventArgs e)
         {
-            await this.JoinableTaskFactory.SwitchToMainThreadAsync();
+            JoinableTaskFactory.RunAsync(HandleOpenSolutionAsync).Task.LogAndForget(GetType());
+        }
+
+        private async Task HandleOpenSolutionAsync()
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
You should pass a `CancellationToken` to this `SwitchToMainThreadAsync()` call. You'll get one from your `AsyncPackage.InitializeAsync` method, which has one handy.

Reply:
No, the cancellation dialog doesn't come up automatically. 
The advantage of doing this is that if VS is shutting down while you're waiting for the UI thread, you won't hold up the shutdown process, but instead this will throw an `OperationCanceledException`, allowing you to skip the UI thread, exit your method, and ultimately unblock VS exiting more quickly.